iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/RoleBasedAccessControlCheck.java

```diff
@@ -19,70 +19,61 @@
+import org.sonar.iac.terraform.checks.utils.ExpressionPredicate;
+import org.sonar.iac.terraform.symbols.AttributeSymbol;
+import org.sonar.iac.terraform.symbols.BlockSymbol;
+
+import static org.sonar.iac.terraform.checks.utils.ExpressionPredicate.isFalse;
+import static org.sonar.iac.terraform.plugin.TerraformProviders.Provider.Identifier.AZURE;
 
 @Rule(key = "S6383")
-public class RoleBasedAccessControlCheck extends AbstractResourceCheck {
+public class RoleBasedAccessControlCheck extends AbstractNewResourceCheck {
 
   private static final String MISSING_MESSAGE = "Omitting '%s' disables role-based access control for this resource. Make sure it is safe here.";
   private static final String DISABLED_MESSAGE = "Make sure that disabling role-based access control is safe here.";
 
+  private static final Version AZURE_V_3 = Version.create(3, 0);
+
   @Override
-  protected void registerResourceChecks() {
-    register(RoleBasedAccessControlCheck::checkKubernetesCluster, "azurerm_kubernetes_cluster");
-    register(RoleBasedAccessControlCheck::checkKeyVault, "azurerm_key_vault");
-  }
+  protected void registerResourceConsumer() {
+    // Report if 'role_based_access_control' block is missing or check its attributes.
+    // Report if 'role_based_access_control.enabled' is set to false
+    // or check 'role_based_access_control.azure_active_directory' block
+    // Report if 'role_based_access_control.azure_active_directory.managed' is set to true
+    // and 'role_based_access_control.azure_active_directory.managed' is missing or set to false.
+    // https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster
+    register("azurerm_kubernetes_cluster", resource -> {
+      BlockSymbol rbac = resource.block("role_based_access_control");
 
-  /**
-   * Report if 'role_based_access_control' block is missing or check its attributes
-   */
-  private static void checkKubernetesCluster(CheckContext ctx, BlockTree resource) {
-    PropertyUtils.get(resource, "role_based_access_control", BlockTree.class)
-      .ifPresentOrElse(rbac -> checkRoleBasedAccessControl(ctx, rbac),
-        () -> reportResource(ctx, resource, String.format(MISSING_MESSAGE, "role_based_access_control")));
-  }
+      if (resource.provider(AZURE).hasVersionLowerThan(AZURE_V_3)) {
+        rbac.reportIfAbsent(MISSING_MESSAGE);
+      }
 
-  /**
-   * Report if 'role_based_access_control->enabled' is set to false
-   * or check 'role_based_access_control->azure_active_directory' block
-   */
-  private static void checkRoleBasedAccessControl(CheckContext ctx, BlockTree rbac) {
-    PropertyUtils.get(rbac, "enabled", AttributeTree.class)
-      .filter(attr -> TextUtils.isValueFalse(attr.value()))
-      .ifPresentOrElse(attr -> ctx.reportIssue(attr, DISABLED_MESSAGE),
-        () -> checkAzureActiveDirectory(ctx, rbac));
-  }
+      AttributeSymbol rbacEnabled = rbac.attribute("enabled").reportIf(isFalse(), DISABLED_MESSAGE);
```

Comment:
Should we now also check `azure_active_directory_role_based_access_control.azure_rbac_enabled`?